docker/bleeding-edge-compilers/Dockerfile

```diff
@@ -1,28 +1,62 @@
 FROM ubuntu:lunar
 
-# Install OS depdencies
-RUN apt-get update -y && apt-get install wget git curl xz-utils build-essential libmpc-dev jq gcc-multilib software-properties-common -y
+USER root
 
-# Set our working dir to /tmp
-WORKDIR /tmp
+ARG PUID
+ARG PGID
 
-# Download and install the latest version of LLVM and CLANG
-RUN LATEST_RELEASE_ARCHIVE=$(curl -s https://api.github.com/repos/llvm/llvm-project/releases/latest | grep browser_download_url | cut -d\" -f4 | egrep 'clang%2Bllvm-(.*?)-x86_64-linux-gnu-ubuntu-22.04.tar.xz') && \
-    wget -O llvm-clang-latest.tar.xz $LATEST_RELEASE_ARCHIVE && \
-    tar -xvf llvm-clang-latest.tar.xz && mv "clang+llvm"* llvm-clang-latest && \
-    LATEST_RELEASE=$(echo $LATEST_RELEASE_ARCHIVE | grep -Eo '[0-9][0-9]\.[0-9]\.[0-9]+' | head -1) && \
-    major=$(echo $LATEST_RELEASE | cut -d. -f1) && \
-    echo "${major}" > clang-major-version.txt && \
-    rm -rf llvm-clang-latest.tar.xz
+ENV TZ="America/New_York"
 
-# Add latest clang,llvm to PATH
-ENV PATH="${PATH}:/tmp/llvm-clang-latest/bin"
+# Install OS depdencies
+RUN apt-get update -y && apt-get install wget git curl xz-utils build-essential libmpc-dev jq gcc-multilib software-properties-common cmake make ninja-build sudo dos2unix python3 python3-pip python3-venv \
+    libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget  \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev -y
+
+RUN ln -sf python3 /usr/bin/python
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 10
 
 # Install latest gcc, g++ from ubuntu toolchain ppa
 # https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test
 # https://gcc.gnu.org/releases.html
 RUN LATEST_RELEASE=$(curl https://api.github.com/repos/gcc-mirror/gcc/tags | jq '.[]|select(.name | startswith("releases/gcc-"))' | jq '.name' | head -1 | grep -Eo '[0-9][0-9]\.[0-9]\.[0-9]+') && \
     major=$(echo $LATEST_RELEASE | cut -d. -f1) && \
     add-apt-repository ppa:ubuntu-toolchain-r/test -y && apt-get update -y && apt-get install gcc-${major} g++-${major} -y && \
-    echo "${major}" > gcc-major-version.txt 
+    echo "${major}" > gcc-major-version.txt && cat gcc-major-version.txt && \
+    # Set gcc and g++ to latest version
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-"${major}" "${major}" && \
+    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-"${major}" "${major}"
+
+# Add Jenkins user and group
+RUN addgroup -gid ${PGID} jenkins
+RUN adduser --disabled-password --gecos '' --ingroup jenkins --uid ${PUID} jenkins
+
+RUN chown -R jenkins:jenkins /usr/local
+RUN chown -f -R jenkins:jenkins /etc || true
+
+USER jenkins
+WORKDIR /home/jenkins
+
+ENV clangbuild=/home/jenkins/llvm-project/build
+ENV PATH=$clangbuild/bin:$PATH
+ENV LD_LIBRARY_PATH=$clangbuild/lib:$LD_LIBRARY_PATH
+
+# Download the source code, build & install libcxx
+RUN git clone --depth=1 https://github.com/llvm/llvm-project.git && \

```

Comment:
I don't think we want to do this, since this will download/build un-released versions of clang (right now in the docker image, `clang --version` reports 17.0.0, which doesn't exist yet)

Reply:
Getting the latest official release now.